src-test/src/org/openbravo/test/accounting/PostedNoDocConfigTest.java

```diff
@@ -84,66 +81,156 @@ public void setUp() throws Exception {
+   */
   @Test
   public void testCountWithoutDocTypeConfigured() {
+    InternalConsumption internalConsumption = null;
+    Table materialManagementConsumptionTable = null;
     try {
-      InternalConsumption internalConsumption = createInternalConsumption();
-
-      createInternalConsumptionLine(internalConsumption);
-
-      processInternalConsumption(internalConsumption);
+      internalConsumption = createHeaderAndMaterialManagementConsumptionLines();
+      processInternalConsumption(internalConsumption, "CO");
+      OBDal.getInstance().refresh(internalConsumption);
 
       activeOrDeactiveMaterialManagementConsumptionTable(true);
-
       TestCostingUtils.runCostingBackground();
 
-      OBDal.getInstance().commitAndClose();
-
-      String result = postDocument(internalConsumption);
+      materialManagementConsumptionTable = getMaterialManagementConsumptionTable(internalConsumption);
+      String result = postDocument(internalConsumption, materialManagementConsumptionTable);
+      OBDal.getInstance().refresh(internalConsumption);
 
       assertEquals(OBMessageUtils.messageBD(DB_MESSAGE_NAME), result);
-      assertEquals("N", internalConsumption.getPosted());
+      assertEquals("DT", internalConsumption.getPosted());
     } catch (Exception e) {
       log4j.error(e.getMessage(), e);
       fail(e.getMessage());
+    } finally {
+      if (internalConsumption != null) {
+        OBDal.getInstance().refresh(internalConsumption);
+        if (StringUtils.equals("CO", internalConsumption.getStatus())) {
+          processInternalConsumption(internalConsumption, "VO");
+        } else {
+          OBDal.getInstance().remove(internalConsumption);
+        }
+        OBDal.getInstance().flush();
+      }
     }
   }
 
+  /**
+   * Test to ensure that an OBException is thrown when posting an internal consumption document
+   * with a document type that is later removed. The test handles document creation, processing,
+   * and posting, and it verifies that the proper exception is raised when the document type is missing.
+   * <p>
+   * Steps:
+   * 1. Create and process an internal consumption document with material management lines.
+   * 2. Activate the material management consumption table.
+   * 3. Create a document type and associate it with the internal consumption.
+   * 4. Run the costing background and attempt to post the document.
+   * 5. Verify the posting status and remove the document type.
+   * 6. Attempt to delete accounting records and expect an OBException due to the missing document type.
+   * 7. In the finally block, ensure proper cleanup of the internal consumption and document type.
+   * <p>
+   * Expected Result:
+   * - The document should post with a status of "Y".
+   * - An OBException with the message "@NoDocTypeForDocument@" should be thrown when the document type is missing.
+   */
   @Test
   public void testDiscountWithoutDocTypeConfigured() {
+    String docTypeId = null;
```

Comment:
```suggestion
    String docTypeId = "";
```

Reply:
I will wait for the other reviewers to mention all the corrections and then commit them in a single commit, however, from what you mention I do not see that the null assignment is a problem, however it is true that a correction must be made in the finally block so that the null validation is using StringUtils, specifically on line 178.